effekt/shared/src/main/scala/effekt/Namer.scala

```diff
@@ -877,6 +902,19 @@ object Namer extends Phase[Parsed, NameResolved] {
       case Nil => "{}"
       case many => many.mkString("{", ", ", "}")
     }
+
+  /**
+   * Generate a fresh hole id that is unique in the current module.
+   * Ideally, this id should be somewhat stable wrt. edits to the source code.
+   * For this, we use the name of the current top-level definition, disambiguated by a counter that increments in
+   * depth-first order as Namer traverses the syntax tree.
+   */
+  private def freshHoleId: String = {
+    val prefix = currentDefinition.value.map(_.id.name).getOrElse("hole")
```

Comment:
I like the idea of using the function name as the hole name! But why only toplevel definitions?

Reply:
Fair enough, it might be better to collect the stack of function scopes, e.g. for

```scala
def foo(): Unit = {
  def bar() = <>

  bar()
}
```

the hole could be called `foo_bar`

Reply:
Done.